packages/vscode-extension/src/commonlib/azureLogin.ts

```diff
@@ -379,8 +380,11 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
 
   async getStatus(): Promise<LoginStatus> {
     const azureAccount = this.getAzureAccount();
-    // add this to make sure Azure Account Extension has fully initialized
-    await azureAccount.waitForSubscriptions();
+    if (this.checkVersion()) {
+      // add this to make sure Azure Account Extension has fully initialized
+      // this will wait for login finish when version >= 0.10.0, so loggingIn status will be ignored
+      await azureAccount.waitForSubscriptions();
```

Comment:
Will this be a long waiting? Any need to add timeout?

Reply:
Before 0.10.0, this will be quick. And after 0.10.0 this line will not be executed.

Reply:
Before 0.10.0, this will be quick. And after 0.10.0 this line will not be executed.

Reply:
Make sense.
BTW, better to rename `checkVersion` to `isLegacyVersion`. I was confused by the name so thought this line is for 0.10.0...

Reply:
Done.